### PR TITLE
[APPS-3168]- Shake report pod changes to support iOS send feedback Se…

### PR DESCRIPTION
**Pull request by snehaViewGlass**
…ndgrid redesign

### CODE REVIEW

File: ShakeReport/Library/SRReporter.h [hunk 1]
```diff
@@ -12,11 +12,18 @@
 #import <MessageUI/MFMailComposeViewController.h>
 #import "SRReportViewController.h"
 
+@class SRReporter;
```
Comment (tdaniel-vg) [inline ShakeReport/Library/SRReporter.h]: remove

File: ShakeReport/Library/SRReporter.m [hunk 2]
```diff
@@ -66,6 +66,7 @@ - (id)init
         _lastSessionCrashed = [self crashFlag];
         _displayReportComposerWhenShakeDevice = YES;
         _recordsCrashes = YES;
+        _sendFeedback = NO;
```
Comment (tdaniel-vg) [inline ShakeReport/Library/SRReporter.m]: remove

File: ShakeReport/Library/SRReporter.m [hunk 3]
```diff
@@ -129,7 +130,8 @@ - (NSString *)logs
 #pragma mark Report
 - (BOOL)canSendNewReport
 {
-    return !_composerDisplayed && [MFMailComposeViewController canSendMail];
+    //return !_composerDisplayed && [MFMailComposeViewController canSendMail];
```
Comment (tdaniel-vg) [inline ShakeReport/Library/SRReporter.m]: return delegate != nil || !_composerDisplayed && [MFMailComposeViewController canSendMail]

File: ShakeReport/Library/SRReporter.m [hunk 4]
```diff
@@ -148,7 +150,9 @@ - (void)displayReportComposer:(BOOL)attachImage{
         UIWindow *window = [[UIApplication sharedApplication] keyWindow];
         [self presentReportComposer:navController inViewController:window.rootViewController];
     } else {
-        [self showMailComposer:attachImage];
+        //[self showMailComposer:attachImage];
```
Comment (tdaniel-vg) [inline ShakeReport/Library/SRReporter.m]: if (delegate == nil) { [self showMailComposer:attachImage];
else {
 //do your thing
}

File: ShakeReport/Library/SRReporter.m [hunk 5]
```diff
@@ -379,6 +383,27 @@ - (void)showMailComposer {
     [self showMailComposer:YES];
 }
 
+- (void) sendFeedbackMailComposer:(BOOL)attachImage {
```
Comment (tdaniel-vg) [inline ShakeReport/Library/SRReporter.m:386]: no need for this function now

File: ShakeReport/Library/SRReporter.m [hunk 6]
```diff
@@ -379,6 +383,27 @@ - (void)showMailComposer {
...
+    
+    if(_sendFeedback)
+    {
+        [self addAttachmentsToMailSendFeedback:attachImage];
+    }
+}
+
+- (void)addAttachmentsToMailSendFeedback:(BOOL)attachImage
+{
+    NSData *imageData;
```
Comment (tdaniel-vg) [inline ShakeReport/Library/SRReporter.m]: set to nil.

**Review by tdaniel-vg — CHANGES_REQUESTED** [review]

File: ShakeReport/Library/SRReporter.m (hunk 5 above)
Comment (snehaViewGlass) [reply ShakeReport/Library/SRReporter.m:386]: removed the below function

File: ShakeReport/Library/SRReporter.h (hunk 1 above)
Comment (snehaViewGlass) [reply ShakeReport/Library/SRReporter.h]: removed

File: ShakeReport/Library/SRReporter.m (hunk 3 above)
Comment (snehaViewGlass) [reply ShakeReport/Library/SRReporter.m]: changed this

File: ShakeReport/Library/SRReporter.m (hunk 6 above)
Comment (snehaViewGlass) [reply ShakeReport/Library/SRReporter.m]: set to nil

File: ShakeReport/Library/SRReporter.m (hunk 4 above)
Comment (snehaViewGlass) [reply ShakeReport/Library/SRReporter.m]: changed it

File: ShakeReport/Library/SRReporter.m (hunk 2 above)
Comment (snehaViewGlass) [reply ShakeReport/Library/SRReporter.m]: removed

File: ShakeReport/Library/SRReporter.m [hunk 7]
```diff
@@ -129,7 +129,7 @@ - (NSString *)logs
 #pragma mark Report
 - (BOOL)canSendNewReport
 {
-    return !_composerDisplayed && [MFMailComposeViewController canSendMail];
+    return _delegate!= nil || (!_composerDisplayed && [MFMailComposeViewController canSendMail]);
```
Comment (tdaniel-vg) [inline ShakeReport/Library/SRReporter.m]: _delegate<space> != nil

**Review by tdaniel-vg — APPROVED** [review]

File: ShakeReport/Library/SRReporter.m (hunk 7 above)
Comment (snehaViewGlass) [reply ShakeReport/Library/SRReporter.m]: changed it

**Pull request merged**